src/main.c

```diff
@@ -380,6 +394,11 @@ int main(int argc, char **argv)
 
 	if (run_tunnel(&cfg) == 0)
 		ret = EXIT_SUCCESS;
+	while ((ret == EXIT_SUCCESS) && (cfg.loop!=0) && (get_sig_received()==0)) {
+		sleep(cfg.loop);
+		if (run_tunnel(&cfg) != 0)
+			ret = EXIT_FAILURE;
+	}
```

Comment:
Less code, but I don't know if it is readable. What do you think?
```
	ret = EXIT_SUCCESS;
	while ((ret == EXIT_SUCCESS) && (cfg.loop!=0) && (get_sig_received()==0)) {
		if (run_tunnel(&cfg) != 0)
			ret = EXIT_FAILURE;
		else:
			sleep(cfg.loop);
	}
```
or:
```
	while ((cfg.loop!=0) && (get_sig_received()==0)) {
		if (run_tunnel(&cfg) != 0) {
			ret = EXIT_FAILURE;
			break;
		}
		sleep(cfg.loop);
	}
```

Reply:
yes, I was looking for something like this, but first didn't get the signal handling right and after I had introduced `get_sig_received()` I forgot to come back to this loop condition. Something with `do { ... } while ( )` might be another solution that doesn't need a `break;`

Reply:
Right, `do { ... } while ( )` might be better. But don't we need a break anyway to avoid the last `sleep()` call which should occur after the `run_tunnel()` call?

Reply:
I have just updated the branch with https://github.com/adrienverge/openfortivpn/pull/190/commits/d489dd7d22eb3821c5dc283afd20529fe7385325 which addresses this (not sure if my `if`s are nicer than your `break`s though...)